[kohkaixun] iP

kohkaixun opened the pull request:
# **Duke ChatBot**
> "[If you fail to plan, you are planning to fail.](https://www.goodreads.com/quotes/460142-if-you-fail-to-plan-you-are-planning-to-fail)" -Benjamin Franklin

Do you
- [ ] have a lot of assignment deadlines to keep track of?
- [ ] have tons of group meetings to schedule?
- [ ] feel overwhelmed by the stuff you have to remember?

If you ticked off at least one of those boxes, fret not! After all, who has the time to keep track about such things? Duke is here to save you that trouble! It's
- easy
- convenient
- and most importantly, ~~$9.99 a month~~ ***FREE***

And it's just a few clicks away! Just
1. download it
2. double click it
3. add the tasks
4. sit back and relax 😎 
----------------------------------------------------------------------------------------------------------------------------
If you're a Java programmer as well, you're welcome to use the code to practice Java! Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

kohkaixun committed:
Added echo bye exit feature

kohkaixun committed:
Added store display list feature

kohkaixun committed:
Added TaskList and Task class

kohkaixun committed:
Added ToDo, Deadline and Event classes

kohkaixun committed:
Added error handling feature

kohkaixun committed:
Added enum TaskType

kohkaixun committed:
Add A-TextUiTesting

kohkaixun committed:
Add save feature

kohkaixun committed:
Add date time OOP

kohkaixun committed:
Merge branch-Level-7's save feature

kohkaixun committed:
Add date time OOP

kohkaixun committed:
Add more OOP

kohkaixun committed:
Divide classes into packages

kohkaixun committed:
Merge branch 'add-gradle-support'

kohkaixun committed:
Add gradle

kohkaixun committed:
Add JUnit test for TaskList and Event class methods

kohkaixun committed:
Create JAR file

kohkaixun committed:
Add JavaDocs

kohkaixun committed:
Apply coding standard

kohkaixun committed:
Add find feature

kohkaixun committed:
Merge branch 'branch-A-JavaDoc'

kohkaixun committed:
Merge branch 'branch-A-CodingStandard'

kohkaixun committed:
Merge branch 'branch-Level-9'

kohkaixun committed:
Add checkstyle files and correct style

kohkaixun committed:
Merge branch 'branch-A-CheckStyle'

jjiayyingtt commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+     * The main method.
+     * @param args Command line arguments
+     */
+    public static void main(String[] args) {
```
nice and consistent coding style

jjiayyingtt commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,131 @@
+                if (!str.contains(target)) {
+                    throw new RuntimeException("Unable to create Deadline! Deadline commands need a /by field!");
+                }
+                int index = str.indexOf(target);
```
this part is slightly harder to understand, maybe you can create a new method to handle dates.

jjiayyingtt commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,86 @@
+    private String path;
+    private File file;
+
+    /**
```
all public classes and methods have JavaDoc, nice!

nt-nic commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,131 @@
+                    throw new RuntimeException("Unable to create Deadline! "
+                            + "Check your date format. Use / to separate day, month and year.");
+                }
+            } else if (eventCheck) {
```
Perhaps, might consider a simpler method to instantiate event tasks

nt-nic commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,55 @@
+        String expected2 = "event " + expected + "1";
+        assertEquals(expected2, event.taskToSavedForm());
+    }
+}
```
Nice clean style and indentations throughout

kohkaixun committed:
Change Duke to GUI

kohkaixun committed:
Merge branch 'branch-Level-10'

kohkaixun committed:
Add assertions

Assertions in static method makeTask of Task class confirm assumption
that todo, deadline and event commands are of their respective minimum
lengths 6, 29 and 45.

Ensures that deadline and event commands have the respective /by field
and /from and /to fields.

kohkaixun committed:
Merge pull request #1 from kohkaixun/branch-A-Assertions

Add assertions

kohkaixun committed:
Change Parser method to switch case

Parser method parse if else statements separated into several static
methods. The parse method has been changed to switch case for easier
readability.

kohkaixun committed:
Merge pull request #2 from kohkaixun/branch-A-CodeQuality

Change Parser method to switch case

kohkaixun committed:
Merge branch 'master' into branch-A-CodeQuality

kohkaixun committed:
Merge branch 'branch-A-CodeQuality'

kohkaixun committed:
Remove magic values in code

Many instances of magic numbers and strings were removed and replaced
by variables to improve readability.

kohkaixun committed:
Add more assertion

Assertions to verify assumptions of command string added in parse method
of Parser class. Makes sure index of delete, mark and unmark commands
are within the range of 0 and 99 so the corresponding TaskList methods
won't get an index out of range of the array in TaskList.

kohkaixun committed:
Add udpate feature

Created UpdateCommand class which executes update method in TaskList
class. Command format is "update <index> <new description or /by, /from
or /to> <new date and time if previouis field was /by, /from or /to>".

Tasks can now be updated instead of having to delete it and then adding
it with the changes again.

kohkaixun committed:
Change method name to updateTask

Name of update method in TaskList class changed to updateTask.

Specify that this is only updating one task and prevent confusion in
future if need to implement update method that updates entire TaskList
object.

kohkaixun committed:
Merge branch 'branch-BCD-Extension'

kohkaixun committed:
Add Event class methods to EventStub

For compatibility.

kohkaixun committed:
Add user guide

User guide on command instructions and formats updated in README.md in
docs folder.

kohkaixun committed:
Update user guide

Images now using GitHub links instead of relative filepaths.

Images now can be seen on https://kohkaixun.github.io/ip/ since links
are used instead of filepaths.

kohkaixun committed:
Update user guide

kohkaixun committed:
Create JAR file

kohkaixun committed:
Update build.gradle file

kohkaixun committed:
Update Duke
